### DISCUSSION TITLE
mpv播放DTS-HD 5.1降混的疑惑

### DISCUSSION THREAD

**Discussion by redomCL** [hooke007/mpv_PlayKit#417] [mpv 通用]
![QQ截图20240222003922](https://github.com/hooke007/MPV_lazy/assets/22269254/896c8c7c-bc49-486b-9a5d-fe9f7400f8c6)
有些DTS-HD 5.1可以降混，这里会显示2，有些依然会显示6，为什么呢？

**Comment by redomCL** [comment]
所有的flac 6声道也是这个问题

**Comment by hooke007** [comment]
下混不需要其它参数 只需要 audio-channels 日志中会记录详细的转换过程

**Reply by hooke007** [reply]
你看到的声道数问题是因为 ad-lavc-downmix
它的效果实测太简单粗暴且如手册所述，不支持所有情况

**Reply by hooke007** [reply]
FAQ更新了下混相关的措辞。下混后的人声后处理参见相关条目
https://github.com/hooke007/MPV_lazy/wiki/0_FAQ#%E5%A4%9A%E5%A3%B0%E9%81%93%E9%9F%B3%E9%A2%91%E7%9A%84%E4%B8%8B%E6%B7%B7

**Reply by redomCL** [reply]
![`E~7RZ_AAUB3~99FX @ 8%9](https://github.com/hooke007/MPV_lazy/assets/22269254/7eb95e5f-608c-4cb4-9bd5-e17409e5d3ec)
尝试只用audio-channels=stereo，声道数依然显示6，还有哪里设置的不对吗？

**Reply by redomCL** [reply]
日志里不管声道是多少  都会显示AO:....2ch

**Reply by hooke007** [reply]
AO当然是2，你都设置了audio output 2 channel了，问题在哪里？
你图里的声道数是解码端的数据，跟你输出端又没关系
mpv的下混用的是af，你看错地方了

**Reply by redomCL** [reply]
哦，想起来了，之前用了ad-lavc-downmix=yes，所以DTS 7.1声道数在OSD显示2，让我误以为OSD那里显示的输出声道数，问题解决了

**Comment by redomCL** [comment]
使用auto_loudnorm.lua后这几个警告是什么意思？
![QQ截图20240228012444](https://github.com/hooke007/MPV_lazy/assets/22269254/4c196f92-7098-46cf-bba4-6fc6d90bdc11)
![QQ截图20240228012451](https://github.com/hooke007/MPV_lazy/assets/22269254/a2e98d6e-cf28-45b3-a22f-51c650c0ca40)

**Reply by redomCL** [reply]
![0](https://github.com/hooke007/MPV_lazy/assets/22269254/b4dc0b5c-3c24-42d4-ad3c-6367d07064e7)
打扰了，已经查到了，但我还是想知道这是什么原因出警告

**Reply by hooke007** [reply]
原因就是mpv闲的
del一个不存在的标签没必要在info级弹警告

**Reply by redomCL** [reply]
if achannels > 2 then
			mp.commandv("af", "pre", "loudnorm")
		else
			mp.commandv("af", "remove", "loudnorm")

为什么不这样写？这样可以避开mpv警告，因为某些规范这样写不规范吗......m af toggle loudnorm感谢我知道该怎么做了

**Reply by hooke007** [reply]
都是错的
你不考虑两个连着多声道文件。前面那个做法加多了或者删不干净，后边那个做法原地打转。

**Reply by redomCL** [reply]
我干脆快捷键开关好了

**Reply by hooke007** [reply]
除非有开着控制台看视频的习惯。。。我觉得这个spam不是什么大问题

**Reply by redomCL** [reply]
了解了  现在还差下面那个音频设备的警告是什么意思？

**Reply by hooke007** [reply]
目测是跟音频缓冲有关系，推流和玩滤镜经常遇到，耳朵没听出问题我一般不管。

**Reply by hooke007** [reply]
重写脚本了

**Comment by redomCL** [comment]
还有下面那个音频设备警告是什么？感觉像是提示音频设备不满足原始声道数？感觉这也没必要以黄字警告形式

**Comment by redomCL** [comment]
![图片](https://github.com/hooke007/MPV_lazy/assets/22269254/5eba9824-1483-43e8-a9d6-6fe7b9116172)
突然想起来mpc-be的降混，这两个降混区别是什么？分离器分离出来音频后直接降混和分离器分离出来后喂给音频解码器，由音频解码器降混的区别？

**Reply by hooke007** [reply]
这问题你为什么不去问mpc的开发者。。。

**Reply by redomCL** [reply]
因为那边已经问了，暂时还没回复，我在想如果或许你也知道能回答一下我也可以

**Reply by redomCL** [reply]
![QQ截图20240301220834](https://github.com/hooke007/MPV_lazy/assets/22269254/697d5b95-5fa3-4434-aaae-da11cdec0912)
开发者回复我了
